agent/grpc-external/services/resource/watch.go

```diff
@@ -1,16 +1,40 @@
 package resource
 
 import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/consul/acl"
 	"github.com/hashicorp/consul/internal/storage"
 	"github.com/hashicorp/consul/proto-public/pbresource"
 )
 
 func (s *Server) WatchList(req *pbresource.WatchListRequest, stream pbresource.ResourceService_WatchListServer) error {
 	// check type exists
-	if _, err := s.resolveType(req.Type); err != nil {
+	reg, err := s.resolveType(req.Type)
+	if err != nil {
 		return err
 	}
 
+	// check acls
+	// TODO(spatel): FIXME Where do I get the ctx in order to extract the token?
+	ctx := context.Background()
+	authz, err := s.ACLResolver.ResolveTokenAndDefaultMeta(tokenFromContext(ctx), nil, nil)
```

Comment:
Using `stream.Context()` here should do the trick.

Reply:
Fixed in https://github.com/hashicorp/consul/pull/16842/commits/b353ea51da4a42e8af8b3ceca67bfc312b0934ef